core/src/main/scala/kafka/zookeeper/ZooKeeperClient.scala

```diff
@@ -269,21 +311,16 @@ class ZooKeeperClient(connectString: String,
   private def initialize(): Unit = {
     if (!zooKeeper.getState.isAlive) {
       info(s"Initializing a new session to $connectString.")
-      var now = System.currentTimeMillis()
-      val threshold = now + connectionTimeoutMs
-      while (now < threshold) {
+      // retry forever until ZooKeeper can be instantiated
+      while (true) {
         try {
           zooKeeper.close()
           zooKeeper = new ZooKeeper(connectString, sessionTimeoutMs, ZooKeeperClientWatcher)
-          waitUntilConnected(threshold - now, TimeUnit.MILLISECONDS)
           return
         } catch {
-          case _: Exception =>
-            now = System.currentTimeMillis()
-            if (now < threshold) {
-              Thread.sleep(1000)
-              now = System.currentTimeMillis()
-            }
+          case e: Exception =>
+            debug("Error when recreating ZooKeeper", e)
```

Comment:
Would we want this to be `info`? It seems like it should be rare, right?

Reply:
Done.